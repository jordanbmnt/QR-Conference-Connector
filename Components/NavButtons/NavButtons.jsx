import { Button, ButtonText, HStack } from "@gluestack-ui/themed";
import { StyleSheet } from "react-native";
import { useDispatch, useSelector } from "react-redux";
import {
  setPreviousPage,
  setScanViewVisible,
  setViewContact,
} from "../../Redux/appSlice";

export default NavButtons = ({ validity }) => {
  const dispatch = useDispatch();
  const scanViewVisible = useSelector(
    (state) => state.appSlice.scanViewVisible
  );
  const previousPage = useSelector((state) => state.appSlice.previousPage);
  return (
    <HStack>
      <Button
        sx={styles.button}
        onPress={() => {
          if (scanViewVisible) {
            dispatch(setScanViewVisible(false));
          } else {
            dispatch(setScanViewVisible(true));
            dispatch(setPreviousPage(false));
          }
        }}
        isDisabled={!validity}
      >
        <ButtonText>
          {scanViewVisible ? "Contacts" : "Tap to Scan Again"}
        </ButtonText>
      </Button>
      {previousPage && scanViewVisible && (
        <Button
          sx={styles.button}
          onPress={() => {
            if (previousPage) {
              dispatch(setScanViewVisible(false));
              dispatch(setViewContact(true));
            }
          }}
          isDisabled={!validity}
        >
          <ButtonText>Back</ButtonText>
        </Button>
      )}
    </HStack>
  );
};

const styles = StyleSheet.create({
  button: {
    marginTop: 0,
    margin: 30,
    marginBottom: 40,
  },
});
