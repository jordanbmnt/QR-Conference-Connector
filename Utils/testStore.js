import { render } from "@testing-library/react-native";
import { Provider } from "react-redux";
import { setupUserInfoStore } from "../Redux/store";
import { GluestackUIStyledProvider } from "@gluestack-ui/themed";
import { config } from "@gluestack-ui/config";

export const initialState = {
  "appSlice": {
    viewContactInfo: false,
  scanViewVisible: true,
  previousPage: false,
  contactInfo: {},
  alertInfo: {
    visibility: false,
    theme: "",
  }
  },
};

export function renderWithProviders(
  ui,
  {
    preloadedState = initialState,
    store = setupUserInfoStore(preloadedState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <GluestackUIStyledProvider config={config}>
          {children}
        </GluestackUIStyledProvider>
      </Provider>
    );
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
