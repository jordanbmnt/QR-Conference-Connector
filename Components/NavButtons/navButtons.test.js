import NavButtons from "./NavButtons";
import { renderWithProviders } from "../../Utils/testStore";

describe("NavButtons", function () {
  const initialState = {
    appSlice: {
      scanViewVisible: true,
      previousPage: false,
    },
  };
  const showBackButton = {
    appSlice: {
      scanViewVisible: true,
      previousPage: true,
    },
  };

  it("should display a list of two buttons (button element and text)", function () {
    const navButtons = renderWithProviders(<NavButtons validity={true} />, {
      preloadedState: initialState,
    });
    const buttons = navButtons.getAllByRole({ role: "button" });
    expect(navButtons).toMatchSnapshot();
    expect(buttons.length).toBe(2);
  });

  it("should display a list of four buttons (button element and text) when the Scan View is visible and Previous Page exists", function () {
    const navButtons = renderWithProviders(<NavButtons validity={true} />, {
      preloadedState: showBackButton,
    });
    const buttons = navButtons.getAllByRole({ role: "button" });
    expect(navButtons).toMatchSnapshot();
    expect(buttons.length).toBe(4);
  });
});
