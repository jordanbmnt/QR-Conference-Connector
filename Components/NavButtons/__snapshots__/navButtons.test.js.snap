// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavButtons should display a list of four buttons (button element and text) when the Scan View is visible and Previous Page exists 1`] = `
<View
  data-style=""
  dataSet={
    {
      "style": "",
    }
  }
  style={
    [
      undefined,
      {},
      {
        "flexDirection": "row",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    action="primary"
    collapsable={false}
    data-style=""
    dataSet={
      {
        "style": "",
      }
    }
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="button"
    size="md"
    style={
      [
        undefined,
        {},
        {
          "alignItems": "center",
          "backgroundColor": "#0077E6",
          "borderRadius": 4,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
        {
          "backgroundColor": "#0077E6",
          "borderColor": "#4AA9FF",
        },
        {},
        {
          "height": 40,
          "paddingHorizontal": 20,
        },
        {},
        {
          "backgroundColor": "#0077E6",
          "borderColor": "#4AA9FF",
        },
        {},
        {
          "height": 40,
          "paddingHorizontal": 20,
        },
        {},
        {
          "margin": 30,
          "marginBottom": 40,
          "marginTop": 0,
        },
        undefined,
      ]
    }
    sx={
      {
        "margin": 30,
        "marginBottom": 40,
        "marginTop": 0,
      }
    }
    variant="solid"
  >
    <Text
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      size="md"
      style={
        [
          undefined,
          {},
          {
            "color": "#525252",
            "fontFamily": undefined,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          undefined,
          {
            "fontSize": 16,
            "lineHeight": 22,
          },
          undefined,
          {},
          {
            "color": "#FCFCFC",
          },
          undefined,
          {
            "color": "#FCFCFC",
            "fontWeight": "600",
          },
          {
            "color": "#005DB4",
          },
          {
            "color": "#FCFCFC",
          },
          {},
          {
            "color": "#FCFCFC",
          },
          undefined,
        ]
      }
    >
      Contacts
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    action="primary"
    collapsable={false}
    data-style=""
    dataSet={
      {
        "style": "",
      }
    }
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="button"
    size="md"
    style={
      [
        undefined,
        {},
        {
          "alignItems": "center",
          "backgroundColor": "#0077E6",
          "borderRadius": 4,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
        {
          "backgroundColor": "#0077E6",
          "borderColor": "#4AA9FF",
        },
        {},
        {
          "height": 40,
          "paddingHorizontal": 20,
        },
        {},
        {
          "backgroundColor": "#0077E6",
          "borderColor": "#4AA9FF",
        },
        {},
        {
          "height": 40,
          "paddingHorizontal": 20,
        },
        {},
        {
          "margin": 30,
          "marginBottom": 40,
          "marginTop": 0,
        },
        undefined,
      ]
    }
    sx={
      {
        "margin": 30,
        "marginBottom": 40,
        "marginTop": 0,
      }
    }
    variant="solid"
  >
    <Text
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      size="md"
      style={
        [
          undefined,
          {},
          {
            "color": "#525252",
            "fontFamily": undefined,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          undefined,
          {
            "fontSize": 16,
            "lineHeight": 22,
          },
          undefined,
          {},
          {
            "color": "#FCFCFC",
          },
          undefined,
          {
            "color": "#FCFCFC",
            "fontWeight": "600",
          },
          {
            "color": "#005DB4",
          },
          {
            "color": "#FCFCFC",
          },
          {},
          {
            "color": "#FCFCFC",
          },
          undefined,
        ]
      }
    >
      Back
    </Text>
  </View>
</View>
`;

exports[`NavButtons should display a list of two buttons (button element and text) 1`] = `
<View
  data-style=""
  dataSet={
    {
      "style": "",
    }
  }
  style={
    [
      undefined,
      {},
      {
        "flexDirection": "row",
      },
      {},
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    action="primary"
    collapsable={false}
    data-style=""
    dataSet={
      {
        "style": "",
      }
    }
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="button"
    size="md"
    style={
      [
        undefined,
        {},
        {
          "alignItems": "center",
          "backgroundColor": "#0077E6",
          "borderRadius": 4,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        {},
        {
          "backgroundColor": "#0077E6",
          "borderColor": "#4AA9FF",
        },
        {},
        {
          "height": 40,
          "paddingHorizontal": 20,
        },
        {},
        {
          "backgroundColor": "#0077E6",
          "borderColor": "#4AA9FF",
        },
        {},
        {
          "height": 40,
          "paddingHorizontal": 20,
        },
        {},
        {
          "margin": 30,
          "marginBottom": 40,
          "marginTop": 0,
        },
        undefined,
      ]
    }
    sx={
      {
        "margin": 30,
        "marginBottom": 40,
        "marginTop": 0,
      }
    }
    variant="solid"
  >
    <Text
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      size="md"
      style={
        [
          undefined,
          {},
          {
            "color": "#525252",
            "fontFamily": undefined,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {},
          {
            "fontSize": 16,
            "lineHeight": 22,
          },
          undefined,
          {},
          {
            "color": "#FCFCFC",
          },
          {},
          {
            "color": "#FCFCFC",
            "fontWeight": "600",
          },
          {
            "color": "#005DB4",
          },
          {
            "color": "#FCFCFC",
          },
          {},
          {
            "color": "#FCFCFC",
          },
          undefined,
        ]
      }
    >
      Contacts
    </Text>
  </View>
</View>
`;
